core/src/main/java/hudson/diagnosis/OldDataMonitor.java

```diff
@@ -218,19 +216,28 @@ public static void report(Saveable obj, Collection<Throwable> errors) {
             return;
         }
         OldDataMonitor odm = get(j);
-        synchronized (odm) {
-            SaveableReference ref = referTo(obj);
+        SaveableReference ref = referTo(obj);
+        while (true) {
             VersionRange vr = odm.data.get(ref);
-            if (vr != null) vr.extra = buf.toString();
-            else            odm.data.put(ref, new VersionRange(null, buf.toString()));
+            if (vr != null) {
+                vr.extra = buf.toString();
+                break;
+            } else if (odm.data.put(ref, new VersionRange(null, buf.toString())) == null) {
+                break;
+            }
         }
     }
 
     public static class VersionRange {
         private static VersionNumber currentVersion = Jenkins.getVersion();
 
-        VersionNumber min, max;
+        @Restricted(NoExternalUse.class)
+        VersionNumber min;
+        @Restricted(NoExternalUse.class)
+        VersionNumber max;
+        @Restricted(NoExternalUse.class)
         boolean single = true;
+        @Restricted(NoExternalUse.class)
```

Comment:
:ant: No <code>extra</code> getter method, which can be used as a replacement in plugins. The same thing is for <code>single</code>, but it was not public at least